source/book/include/marlin/book/BookStore.h

```diff
@@ -1,102 +1,129 @@
 #pragma once
 
 // -- std includes
+#include <atomic>
+#include <filesystem>
 #include <memory>
+#include <stdexcept>
 #include <string>
 #include <string_view>
-#include <unordered_map>
+#include <thread>
 #include <typeindex>
 #include <typeinfo>
-#include <stdexcept>
+#include <unordered_map>
 
 // -- Marlin includes
 #include "marlin/Exceptions.h"
 
 // -- MarlinBook includes
-#include "marlin/book/ROOTAdapter.h"
-#include "marlin/book/Entry.h"
 #include "marlin/book/Condition.h"
+#include "marlin/book/Entry.h"
 #include "marlin/book/EntryData.h"
 #include "marlin/book/Flags.h"
-#include "marlin/book/Hist.h"
 #include "marlin/book/MemLayout.h"
+#include "marlin/book/ROOTAdapter.h"
 #include "marlin/book/Selection.h"
 
 namespace marlin {
   /// contains classes needed to book and managed booked objects.
   namespace book {
 
-    // -- MarlinBook forward declaration
-    template < unsigned long long >
-    class Flag_t ;
-    class MemLayout ;
-    template < typename, typename... >
-    class SingleMemLayout ;
-    template < typename T,
-               void( const std::shared_ptr< T > &,
-                     const std::shared_ptr< T > & ),
-               typename... >
-    class SharedMemLayout ;
-    class Condition ;
-    class Selection ;
+    template < typename T >
+    class Manager {} ;
 
-    /**
-     *  @brief Base Class for Entry Data, for similar behavior.
-     */
-    template < class T >
-    class EntryDataBase {} ;
+    template < typename T >
+    class Handle< Manager< T > > {
```

Comment:
Entry is meant to be stored in a field, so it can't be a template, 
In my opinion there are 2 options:
1. rename Entry -> ??, and Manager -> Entry
2. rename Entry to something better.